distribution/lib/Standard/Base/0.0.0-dev/src/Data/Range.enso

```diff
@@ -286,14 +290,16 @@ type Range
 
    Arguments:
    - n: The end of the range.
+   - step: The step between consecutive elements of the range. It must be
+     non-zero. Defaults to 1.
 
    > Example
      Create a range containing the numbers 0, 1, 2, 3, 4.
 
          0.up_to 5
-Integer.up_to : Integer -> Range
-Integer.up_to self n = case n of
-    _ : Integer -> Range_Data self n
+Integer.up_to : Integer -> Integer -> Range
+Integer.up_to self n step=1 = case n of
+    _ : Integer -> Range.Value self n step
     _ -> Error.throw (Illegal_Argument_Error_Data "Expected range end to be an Integer.")
```

Comment:
Are you sure about this change?

IMHO it makes the code less readable.

Looking at `0.up_to 10 2` it isn't really clear that `2` is the step. At first it looks like a typo to me.

With named arguments `0.up_to 10 step=2` is a bit better, but we cannot force our users to name the args unfortunately.

Still, IMO the old style of `0.up_to 10 . with_step 2` is a bit clearer.

As for the internal constructs, I don't see why we shouldn't just keep `Range.Value start end step` - for internal usage this is more direct and IMO more readable than `start.up_to end step`. But if you don't want to switch to `Range.Value`, let's at least make the argument named.

Reply:
ok will revert to the with_step.